Sources/Widgets/Widgets3D/PaintWidget/index.js

```diff
@@ -41,24 +44,34 @@ function widgetBehavior(publicAPI, model) {
       model.activeState &&
       model.activeState.getActive()
     ) {
+      const normal = model.camera.getDirectionOfProjection();
+      const up = model.camera.getViewUp();
+      const right = vec3.create();
```

Comment:
Each time I see a `{glmatrix}.create()`, I worry that we will lose precision and we could get bad interaction with models that are far from the [0,0,0] origin. It might not be an issue here, but in general, *never* use the `{glMatrix}.create()` but instead provide the array itself `[0,0,0]` or `new Float64Array(3)` as by default glMatrix will do `new Float32Array(size) + identity(for matrix)`.

So ideally do a search and replace for those `.create()` calls.

Reply:
sounds good, will do.

Reply:
replaced by `[]`